cpp/src/arrow/python/pyarrow.h

```diff
@@ -71,6 +77,8 @@ DECLARE_WRAP_FUNCTIONS(tensor, Tensor)
 DECLARE_WRAP_FUNCTIONS(batch, RecordBatch)
 DECLARE_WRAP_FUNCTIONS(table, Table)
 
+DECLARE_WRAP_FUNCTIONS(extension_id_registry, engine::ExtensionIdRegistry)
```

Comment:
This exposure is for PyArrow. It will be used in an upcoming PR, which should not be merged into this one. The purpose of exposing `ExtensionIdRegistry` is to allow use of a [nested registry](https://github.com/apache/arrow/pull/13232) from PyArrow.

Reply:
This doesn't really address the following comment:
> Instead, you should wrap/unwrap purely on the Cython side, like for most other C++ classes.